sdk/keyvault/azure-security-keyvault-administration/src/test/java/com/azure/security/keyvault/administration/KeyVaultCredentialPolicyTest.java

```diff
@@ -146,16 +183,94 @@ public void onChallengeCredentialPolicy() {
     public void onAuthorizeRequestChallengeCachePresent() {
         KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(this.credential, false);
 
+        StepVerifier.create(onChallengeAndClearCache(policy, this.callContext, unauthorizedHttpResponseWithHeader) // Challenge cache created
+                .then(policy.authorizeRequest(this.testContext))) // Challenge cache used
+            .verifyComplete();
+
+        String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(AUTHORIZATION);
+        assertFalse(tokenValue.isEmpty());
+        assertTrue(tokenValue.startsWith(BEARER));
+    }
+
+    @Test
+    public void onAuthorizeRequestChallengeCachePresentSync() {
+        KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(this.credential, false);
+
         // Challenge cache created
-        onChallenge(policy, this.callContext, unauthorizedHttpResponseWithHeader).block();
+        onChallengeAndClearCacheSync(policy, this.callContext, unauthorizedHttpResponseWithHeader);
         // Challenge cache used
-        policy.authorizeRequest(this.testContext).block();
+        policy.authorizeRequestSync(this.testContext);
 
-        String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(HttpHeaderName.AUTHORIZATION);
+        String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(AUTHORIZATION);
         assertFalse(tokenValue.isEmpty());
         assertTrue(tokenValue.startsWith(BEARER));
     }
 
+    @Test
+    public void onAuthorizeRequestChallengeCachePresentWithClaims() {
+        KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(this.credential, false);
+
+        StepVerifier.create(policy.authorizeRequestOnChallenge(this.callContext, // Challenge cache created
+                    this.unauthorizedHttpResponseWithHeader)
+                .flatMap(result -> result ? policy.authorizeRequestOnChallenge(this.callContext, // Challenge with claims received
+                    this.unauthorizedHttpResponseWithHeaderAndClaims) : Mono.just(false)))
+            .assertNext(result -> {
+                assertTrue(result);
+
+                String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(AUTHORIZATION);
+
+                assertFalse(tokenValue.isEmpty());
+                assertTrue(tokenValue.startsWith(BEARER));
```

Comment:
It appears this test would pass if the policy used the same token before and after the claims challenge. We need to ensure that after a CAE challenge, it requests a token with the specified claims and retries the request, authorized with the new token